chainio/clients/wallet/privatekey_wallet_test.go

```diff
@@ -0,0 +1,70 @@
+package wallet
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	"github.com/Layr-Labs/eigensdk-go/testutils"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	chainId = big.NewInt(31337)
+)
+
+func TestPrivateKeyWallet(t *testing.T) {
+	logger := testutils.NewTestLogger()
+
+	t.Run("SendTransaction + GetTransactionReceipt", func(t *testing.T) {
+		anvilC, err := testutils.StartAnvilContainer("")
+		require.NoError(t, err)
+		ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		anvilHttpEndpoint, err := anvilC.Endpoint(ctxWithTimeout, "http")
+		require.NoError(t, err)
+		ethClient, err := ethclient.Dial(anvilHttpEndpoint)
+		require.NoError(t, err)
+
+		ecdsaPrivKeyHex := "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+		ecdsaPrivKey, err := crypto.HexToECDSA(ecdsaPrivKeyHex)
+		require.NoError(t, err)
+		signerV2, signerAddr, err := signerv2.SignerFromConfig(signerv2.Config{PrivateKey: ecdsaPrivKey}, chainId)
+		if err != nil {
+			panic(err)
+		}
+
+		skWallet, err := NewPrivateKeyWallet(ethClient, signerV2, signerAddr, logger)
+		require.NoError(t, err)
+
+		tx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainId,
+			Nonce:     0,
+			GasTipCap: big.NewInt(1),
+			GasFeeCap: big.NewInt(1_000_000_000),
+			Gas:       21000,
+			To:        &signerAddr,
+			Value:     big.NewInt(1),
+		})
+		ctxWithTimeout, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		txId, err := skWallet.SendTransaction(ctxWithTimeout, tx)
+		require.NoError(t, err)
+
+		// need to give some time for anvil to process the tx and mine the block
+		// TODO: shall we expose a public WaitForTxReceipt function in the wallet interface, or somewhere else?
```

Comment:
the `GetTransactionReceipt` doesn't wait? How would `WaitForTxReceipt ` method diff from that?

Reply:
No `GetTransactionReceipt` only calls the eth rpc node endpoint and returns error if txreceipt doesn't exist (tx hasnt been mined) yet. https://github.com/Layr-Labs/eigensdk-go/blob/de54bdee8272c3c1059bf7601f195589098ae57f/chainio/clients/wallet/privatekey_wallet.go#L66

Reply:
There's a [waitForReceipt](https://github.com/Layr-Labs/eigensdk-go/blob/samlaf-avs-588-upstream-tx-manager-to-eigensdk-go/chainio/txmgr/simple.go#L116) private method on the simple txmgr. Wondering if we shouldn't make that interface public and directly on the wallet instead? Or the ethClient interface.... not sure...

Reply:
but I am not sure if  `waitForReceipt` is a wallet property. since it's always gonna be available via eth node, no matter how we send? hmm unless we consider fireblock state in this too, then it should be on wallet

Reply:
no fireblocks is irrelevant here I think. waiting for a receipt for a txHash is purely polling an eth node for some amount of time waiting for the txReceipt to exist (to have been mined).
I also think it should be some eth node abstraction layer, but we don't have that currently.

Reply:
ok park it for now. and then we can add it to ethBackend I think